app/src/main/java/de/westnordost/streetcomplete/data/elementfilter/OverpassQueryCreator.kt

```diff
@@ -38,14 +41,16 @@ class OverpassQueryCreator(
     }
 
     private fun EnumSet<ElementsTypeFilter>.toOqlNames(): List<String> = when {
-        containsAll(listOf(NODES, WAYS, RELATIONS)) ->  listOf("nwr")
-        containsAll(listOf(NODES, WAYS)) ->             listOf("nw")
-        containsAll(listOf(WAYS, RELATIONS)) ->         listOf("wr")
-        else -> map { when (it!!) {
-            NODES -> "node"
-            WAYS -> "way"
-            RELATIONS -> "rel"
-        } }
+        containsAll(listOf(NODES, WAYS, RELATIONS)) -> listOf("nwr")
+        containsAll(listOf(NODES, WAYS)) -> listOf("nw")
+        containsAll(listOf(WAYS, RELATIONS)) -> listOf("wr")
```

Comment:
See my first comment, I think it is not as well readable as before.

Reply:
Mentioned in https://github.com/streetcomplete/StreetComplete/pull/3649#issuecomment-1015684100 but it didn't make its way into the summary. Direct answer: https://github.com/streetcomplete/StreetComplete/pull/3582#discussion_r766667403 (in the above comment).